### DISCUSSION TITLE
How to improve taper points?

### DISCUSSION THREAD

**Discussion by niubiworker** [gdsfactory/gdsfactory#3210] [Q&A]
I want to draw a taper structure now, with two end-face widths of 4 µm and 1 µm and a length of 1 mm. However, the structure drawn using the taper function from the basic PDK only generated 4 points. I would like to have more points, but the function does not seem to offer an option like npoints. What should I do?
![image](https://github.com/user-attachments/assets/5cdf9c75-6518-4190-8395-88870c684e8e)

**Comment by sebastian-goeldi** [comment]
I would suggest, either write a quick nonlinear_taper function suitable for your needs or use `gf.path.transition` or a general `gf.Path`. There are likely other options as well.
